frontend/packages/ceph-storage-plugin/src/components/modals/storage-pool-modal/storage-pool-modal.tsx

```diff
@@ -0,0 +1,391 @@
+      isList: false,
+      name: newPoolName,
+    };
+  }, [newPoolName]);
+
+  const [newPool, newPoolLoaded, newPoolLoadError] = useK8sWatchResource<StoragePoolKind>(
+    poolResource,
+  );
+
+  React.useEffect(() => {
+    k8sGet(OCSServiceModel, OCS_INTERNAL_CR_NAME, CEPH_STORAGE_NAMESPACE)
+      .then((cluster: OCSStorageClusterKind) => {
+        setStorageClusterLoading(false);
+        if (cluster.status.phase === 'Ready') {
+          setStorageClusterReady(true);
+        }
+      })
+      .catch((e) => setStorageClusterError(e));
+    if (timer && timeoutClear) {
+      clearTimeout(timer);
+    }
+    if (isSubmit) {
+      if (newPool && newPoolLoaded && newPool?.status?.phase === 'Ready') {
+        setPoolStatus('created');
+        setIsSubmit(false);
+        setTimeClear(true);
+      } else if (newPoolLoaded && newPool?.status?.phase === 'ReconcileFailed') {
+        setPoolStatus('failed');
+        setIsSubmit(false);
+        setTimeClear(true);
+      } else if (newPoolLoaded && newPoolLoadError && newPoolLoadError?.response?.status !== 404) {
+        setPoolStatus('failed');
+        setIsSubmit(false);
+        setTimeClear(true);
+      }
+    }
+  }, [isSubmit, newPool, newPoolLoadError, newPoolLoaded, timeoutClear, timer]);
+
+  const replicaDropdownItems = _.keys(replicas).map((replica) => {
+    return (
+      <DropdownItem
+        key={`replica-${replicas[replica]}`}
+        component="button"
+        id={replicas[replica]}
+        onClick={(e) => setReplicaSize(e.currentTarget.id)}
+      >
+        {`Replica ${replicas[replica]}`}
+      </DropdownItem>
+    );
+  });
+
+  const availableDeviceClasses = cephClusterObj[0]?.status?.storage?.deviceClasses.map((device) => {
+    return (
+      <DropdownItem
+        key={`device-${device?.name}`}
+        component="button"
+        id={device?.name}
+        onClick={(e) => setdeviceClass(e.currentTarget.id)}
+      >
+        {device?.name}
+      </DropdownItem>
+    );
+  });
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    setPoolStatus('progress');
+    event.preventDefault();
+    const poolObj: StoragePoolKind = {
+      apiVersion: apiVersionForModel(CephBlockPoolModel),
+      kind: CephBlockPoolModel.kind,
+      metadata: {
+        name: newPoolName,
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      spec: {
+        deviceClass: deviceClass || '',
+        parameters: {
+          compression_mode: isCompressed ? 'aggressive' : '', // eslint-disable-line @typescript-eslint/camelcase
+        },
+        replicated: {
+          size: Number(replicaSize),
+        },
+      },
+    };
+
+    handlePromise(k8sCreate(CephBlockPoolModel, poolObj))
+      .then(() => {
+        setIsSubmit(true);
+        // The modal will wait for 15 sec to get feedback from Rook
+        const timeoutTimer = setTimeout(() => {
+          setPoolStatus('timeout');
+          setIsSubmit(false);
+        }, 15000);
```

Comment:
15 * SECOND, Constant for SECOND already present.

Reply:
ack